### DISCUSSION TITLE
My Template for every Note

### DISCUSSION THREAD

**Discussion by phibr0** [SilentVoid13/Templater#388] [Templates Showcase]
I am using Obsidian in School, I have Folders for each class and all of them are in german - except of course my English Course. I set the language Key in the Frontmatter to one of these Languages depending on which Folder the Note is in. On opening the File I can type in the Subtag ( `#Course/Topic`) and Topic. 

![image](https://user-images.githubusercontent.com/59741989/135327315-b74d6952-ab09-4dcf-b084-f0ec7fdadfa6.png)

```md
---<%*
//If File is untitled prompt the User to set a Title (Source: https://github.com/SilentVoid13/Templater/discussions/259)
let title = tp.file.title
if (title.startsWith("Untitled")) {
    title = await tp.system.prompt("Title") ?? "Untitled";
    await tp.file.rename(`${title}`);
} %>
date: "<% tp.file.creation_date("dddd Do MMMM YYYY HH:mm") %>"
topic: <% tp.file.cursor(1) %>
tags:
  - <% tp.file.folder() %>/<% tp.file.cursor(1) %>
aliases: []
<%*
//If this File is in my English Folder set the language correctly. I assume it will be German otherwise
//This is mostly for the Dictionary Plugin Integration
if(tp.file.folder() === "Englisch") {
	tR += "language: en-US";
} else {
	tR += "language: de";
} %>
---

# <% tp.file.title %>

<% tp.file.cursor(2) %>

```

**Comment by KennSugiyama** [comment]
Hi! Great template!  Curious, do you use the language tags besides organization?

**Reply by phibr0** [reply]
They are used by the dictionary plugin to switch to the correct language based on my currently opened note

**Comment by sandorex** [comment]
What plugin are you using for the script `<% .. %>` ?

**Reply by phibr0** [reply]
What do you mean? Do you mean the `<%* %>` scripts with JS in them? You can learn about them here: https://silentvoid13.github.io/Templater/commands/execution-command.html

**Reply by sandorex** [reply]
Thank you so much i did not know about that feature! 😄 

EDIT: Im an idiot and did not know that Templater and Templates were seperate plugins and was wondering why it wasnt working 🤣

**Reply by wrgrant** [reply]
@sandorex thanks for this! I didn't realize that either! 😆

**Comment by alwaysusa** [comment]
This is very helpful - thank you!

Question - for me, when I start a new note I do get the system prompt to give it a title, but after filling it in the note is named correctly, but the header is still "Untitled". Is this expected behavior?

**Reply by phibr0** [reply]
I also noticed that, I think it worked back when I posted it here. Haven't had any time to update it yet

**Reply by bhuhnke** [reply]
When you add [[ ]] to link the title it automatically changes the title to what you specified. 
This also helps when you happen to change the title or file, it'll automatically change it everywhere! Learned this trick by Nicole van der Hoeven in her video where she sets up Templater.

I use it like this: 
<%*
  let title = tp.file.title
  if (title.startsWith("Untitled")) {
    title = await tp.system.prompt("Title");
    await tp.file.rename(title);
  } %>

"#" [[<%* tR += title %>]]

The hashtag is in "" so markdown doesn't format it here in github :)

**Reply by alwaysusa** [reply]
That works perfectly! Thank you :)
